bugbug/tools/code_review.py

```diff
@@ -117,9 +138,145 @@ def get_review_request_by_id(self, review_id: int) -> ReviewRequest:
     def get_patch_by_id(self, patch_id: int) -> Patch:
         raise NotImplementedError
 
+    @abstractmethod
+    def get_all_inline_comments(
+        self, comment_filter
+    ) -> Iterable[tuple[int, list[InlineComment]]]:
+        raise NotImplementedError
+
+    def load_patch_by_id(self, diff_id) -> Patch:
+        """Load a patch from local cache if it exists.
+
+        If the patch is not in the local cache it will be requested from the
+        provider and cache it locally.
+
+        Args:
+            diff_id: The ID of the patch.
+
+        Returns:
+            The patch.
+        """
+        try:
+            with open(f"patches/{diff_id}.patch", "r") as f:
+                patch = Patch(f.read())
+        except FileNotFoundError:
+            with open(f"patches/{diff_id}.patch", "w") as f:
+                patch = self.get_patch_by_id(diff_id)
+                f.write(patch.raw_diff)
+
+        return patch
+
+    def get_matching_hunk(
```

Comment:
We should add some tests for this function in a follow-up

Reply:
I filed https://github.com/mozilla/bugbug/issues/4223 as a follow-up